### PR TITLE
Added support for BMM150

### PR DESCRIPTION
**Pull request by cikarnd**
Hi, been toying with the IMU in the M5Stack and modified the IMU library to add basic support for the BMM150 (copied the Bosch stuff).
The M5Stack has four magnets at the M5GO Bottom module, so i removed it. There is still some big offset in ‘x’, around 30 (read at the IMU demo application console).

### CODE REVIEW

**Comment by rojer** [comment]
seems reasonable. @pimvanpelt wdyt?

File: src/mgos_imu_bmm150.c [hunk 1]
```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright 2018 Google Inc.
```
Comment (pimvanpelt) [inline src/mgos_imu_bmm150.c]: ThIs file is not copyright Google, unless the author works for Google. Can you make sure to put the correct holder here?

File: src/mgos_imu_bmm150.c [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+}
+
+/* Following code changes license type.
+Original code has been conveniently modified to fit current environment
+requirements */
+
+/**
+ * Copyright (c) 2020 Bosch Sensortec GmbH. All rights reserved.
+ *
+ * BSD-3-Clause
```
Comment (pimvanpelt) [inline src/mgos_imu_bmm150.c]: Please don't mix license types inline. The code that you're borrowing from other repositories should go into a `third_party/bosch/{src,include}/` directory and added to `mos.yml`.

**Review by pimvanpelt — CHANGES_REQUESTED** [review]
The code and hook into the library looks great, thank you.
I suggest to put the bosch code in a third_party/ directory, which should be reasonably straight forward -- for inspiration take a look at https://github.com/mongoose-os-libs/ili9341-spi/blob/master/mos.yml where I did that for some Adafruit code I reused.

Can you also update the `README.md` please?

File: src/mgos_imu_bmm150.c (hunk 1 above)
Comment (cikarnd) [reply src/mgos_imu_bmm150.c]: done

File: src/mgos_imu_bmm150.c (hunk 2 above)
Comment (cikarnd) [reply src/mgos_imu_bmm150.c]: done
In my mind, keeping those functions static and encapsulated took precedence over the licensing issue. Now I know.

File: third-party/bosch/include/bmm150.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#include "mgos.h"
```
Comment (pimvanpelt) [inline third-party/bosch/include/bmm150.h:37]: missing license+copyright header. Please add.

File: src/mgos_imu_bmm150.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#pragma once
```
Comment (pimvanpelt) [inline src/mgos_imu_bmm150.h:18]: Missing copyright+license header. Please add.

File: src/mgos_imu_bmm150.c [hunk 5]
```diff
@@ -0,0 +1,100 @@
+#include "mgos_imu_bmm150.h"
```
Comment (pimvanpelt) [inline src/mgos_imu_bmm150.c:18]: Missing license+copyright header. please add.

**Review by pimvanpelt — CHANGES_REQUESTED** [review]
you're close on this one -one more round of details (copyright+license headers particularly, and a question about bmm150.h which you put in third-party)

File: third-party/bosch/include/bmm150.h (hunk 3 above)
Comment (cikarnd) [reply third-party/bosch/include/bmm150.h:37]: done

**Comment by cikarnd** [comment]
My pleasure

**Review by pimvanpelt — APPROVED** [review]

**Pull request merged**